api/src/main/java/org/apache/iceberg/transforms/PartitionSpecVisitor.java

```diff
@@ -114,13 +115,10 @@ default T unknown(int fieldId, String sourceName, int sourceId, String transform
    */
   @Deprecated
   static <R> List<R> visit(Schema schema, PartitionSpec spec, PartitionSpecVisitor<R> visitor) {
-    List<R> results = Lists.newArrayListWithExpectedSize(spec.fields().size());
-
-    for (PartitionField field : spec.fields()) {
-      results.add(visit(schema, field, visitor));
-    }
-
-    return results;
+    return spec.fields().stream()
+        .map(field -> visit(schema, field, visitor))
+        .filter(Objects::nonNull)
```

Comment:
I don't think it is right to automatically discard nulls. Those might be part of the result produced by the visitor. And it is not possible to see which field produced the null value.